clang/include/clang/Sema/Sema.h

```diff
@@ -7849,13 +7854,17 @@ class Sema final : public SemaBase {
   /// keepInLifetimeExtendingContext - Pull down InLifetimeExtendingContext
   /// flag from previous context.
   void keepInLifetimeExtendingContext() {
-    if (ExprEvalContexts.size() > 2 &&
-        parentEvaluationContext().InLifetimeExtendingContext) {
-      auto &LastRecord = ExprEvalContexts.back();
-      auto &PrevRecord = parentEvaluationContext();
-      LastRecord.InLifetimeExtendingContext =
-          PrevRecord.InLifetimeExtendingContext;
-    }
+    if (ExprEvalContexts.size() > 2)
```

Comment:
I wonder if this `2` is correct. Pre-existing but it should probably be `1` (ie, if there are more than 1 context, there is a parent. Maybe we should just assert >1 actually
(there is always going a context bot the default init + 1 for the translation unit)

Reply:
Good catch! `ExprEvalContexts.size() > 1` is make sense, but I just removed these 2 functions.